src/CLICommand/parameters_helper.ts

```diff
@@ -277,6 +281,37 @@ export class ParametersHelper {
 		return mapFunc(value);
 	}
 
+	// Parse user input through `URL` constructor. This will throw an error on any
+	// provided url that does not have a `protocol` and `host` or is otherwise invalid
+	public getGateway(): URL {
```

Comment:
Consider renaming to `getOptionalGateway` to keep with the convention we've established elsewhere (e.g. getOptionalWallet).